### PR TITLE
Allow mapping to transform to upper case or lower case.

### PR DESCRIPTION
**Pull request by fbacchella**

### CODE REVIEW

**Comment by sandrobonazzola** [comment]
@fbacchella can you please rebase your patch?

**Comment by fbacchella** [comment]
Done.

**Comment by sandrobonazzola** [comment]
@mwperina ,  @arso can you please review?

File: src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java [hunk 1]
```diff
@@ -29,6 +29,28 @@
 import org.ovirt.engine.extension.aaa.misc.*;
```
Comment (mwperina) [inline src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java:30]: Please don't use wildcard import, better is to declare each required import separately

File: src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java [hunk 2]
```diff
@@ -29,6 +29,28 @@
...
 
 public class MappingExtension implements Extension {
+    
+    private enum ChangeCase {
+        TOLOWER {
+            @Override
+            String transform(String value) {
+                return value.toLowerCase();
```
Comment (mwperina) [inline src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java]: Please handle null value

File: src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java [hunk 3]
```diff
@@ -29,6 +29,28 @@
...
+            }
+        },
+        TOUPPER {
+            @Override
+            String transform(String value) {
+                return value.toUpperCase();
```
Comment (mwperina) [inline src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java]: Please handle null value

File: src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java [hunk 4]
```diff
@@ -44,11 +66,13 @@ public String format(String s) {
         private final Pattern pattern;
         private final String replacement;
         private final boolean mustMatch;
+        private final ChangeCase changecase;
 
-        public RegExFormat(Properties props, String prefix) {
+        public RegExFormat(Properties props, String prefix, ChangeCase changecase) {
```
Comment (mwperina) [inline src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java]: Please use Java coding standards, so `changeCase`

File: src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java [hunk 5]
```diff
@@ -44,11 +66,13 @@ public String format(String s) {
         private final Pattern pattern;
         private final String replacement;
         private final boolean mustMatch;
+        private final ChangeCase changecase;
```
Comment (mwperina) [inline src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java]: Please use Java coding standards, so `changeCase`

File: src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java [hunk 6]
```diff
@@ -186,10 +210,19 @@ private void doMapAuthRecord(ExtMap input, ExtMap output) throws Exception {
     private Format createFormat(Properties props, String prefix) {
         Format ret = null;
         String type = props.getProperty(prefix + "type", "void");
+        String newcase = props.getProperty(prefix + "case", "").toLowerCase();
+        ChangeCase changecase;
```
Comment (mwperina) [inline src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java]: Please use Java coding standards, so `changeCase`

File: src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java [hunk 7]
```diff
@@ -186,10 +210,19 @@ private void doMapAuthRecord(ExtMap input, ExtMap output) throws Exception {
...
+        if("lower".equals(newcase)) {
+            changecase = ChangeCase.TOLOWER;
+        } else if ("upper".equals(newcase)){
+            changecase = ChangeCase.TOUPPER;
+        } else {
+            changecase = ChangeCase.KEEP;
+        }
```
Comment (mwperina) [inline src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java]: Could you please move the conversion logic into the `ChangeCase` to `public static ChangeCase valueOf(String value)` method?

**Review by mwperina — CHANGES_REQUESTED** [review]

File: src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java (hunk 1 above)
Comment (fbacchella) [reply src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java:30]: The wildcard are not coming from my patch, they were already used.

File: src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java (hunk 7 above)
Comment (fbacchella) [reply src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java]: Is throwing IllegalArgumentException ok for you ? Or does it needs to be catched ?

File: src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java (hunk 2 above)
Comment (fbacchella) [reply src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java]: done

File: src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java (hunk 3 above)
Comment (fbacchella) [reply src/main/java/org/ovirt/engine/extension/aaa/misc/mapping/MappingExtension.java]: done

**Comment by sandrobonazzola** [comment]
@fbacchella  can you please rebase on top of master?

**Review by mwperina — APPROVED** [review]
+1

**Pull request merged**